client/components/FormInputComponents/LongText/LongText.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from "react";
+import { FieldProps } from "../types";
+import { FieldError, useFormContext } from "react-hook-form";
+import styles from "./LongText.module.css";
+
+const LongText = ({ field, index }: FieldProps) => {
+  const {
+    register,
+    formState: { errors },
+  } = useFormContext();
+  const [textAreaValue, setTextAreaValue] = useState("");
+  const { required, max_length: maxLength } = field.validations ?? {};
+
+  const fieldError = (
+    errors.answers as { [key: number]: { text?: FieldError } } | undefined
+  )?.[index]?.text;
+
+  const { onChange: registerOnChange, ...registerProps } = register(
+    `answers.${index}.text`,
+    {
+      required: required && "This field is required",
+      maxLength: maxLength && {
+        value: maxLength,
+        message: `Answer has to be less than ${maxLength} characters`,
+      },
+    },
+  );
+
+  const handleTextAreaChange = (
+    event: React.ChangeEvent<HTMLTextAreaElement>,
+    initialHeight: string = "auto",
+  ) => {
+    setTextAreaValue(event.target.value);
+    const textArea = event.target;
+    textArea.style.height = initialHeight;
+    textArea.style.height = `${textArea.scrollHeight}px`; // Set height
+    registerOnChange(event);
+  };
```

Comment:
@elimanzo great work on this event handler. I see that you are recalculating the height based on `scrollHeight` which is fine in theory  since we only ensure the textarea be as tall as needed to fit the content and the content is restricted to a certain size, but should we set maximum height restrictions just in case?
```suggestion
  const handleTextAreaChange = (
  event: React.ChangeEvent<HTMLTextAreaElement>,
  initialHeight: string = "auto",
  maxHeight: string = "500px" // Set a maximum height
) => {
  setTextAreaValue(event.target.value);
  const textArea = event.target;
  textArea.style.height = initialHeight;
  textArea.style.height = `${textArea.scrollHeight}px`;
  if (textArea.scrollHeight > parseInt(maxHeight)) {
    textArea.style.height = maxHeight;
  }
  registerOnChange(event);
};
```

Reply:
Good call!